packages/react-app/src/constants/bountyExportItems.ts

```diff
@@ -0,0 +1,25 @@
+import { BountyCollection } from '@app/models/Bounty';
+
+type bountyKeys = keyof BountyCollection;
+type rewardKeys = keyof BountyCollection['reward'];
+type createdByKeys = keyof BountyCollection['createdBy'];
+type claimedByKeys = keyof BountyCollection['claimedBy'];
+
+export const BOUNTY_EXPORT_ITEMS = [
+	{ label: 'ID', key: ('_id' as bountyKeys) },
+	{ label: 'Title', key: ('title' as bountyKeys) },
+	{ label: 'Description', key: ('description' as bountyKeys) },
+	{ label: 'Criteria', key: ('criteria' as bountyKeys) },
+	{ label: 'Status', key: ('status' as bountyKeys) },
+	{ label: 'Due', key: ('dueAt' as bountyKeys) },
+	{ label: 'Reward Amount', key: ('reward' as bountyKeys) + '.' + ('amount' as rewardKeys) },
+	{ label: 'Reward Currency', key: ('reward' as bountyKeys) + '.' + ('currency' as rewardKeys) },
+	{ label: 'Created', key: ('createdAt' as bountyKeys) },
+	{ label: 'Created By', key: ('createdBy' as bountyKeys) + '.' + ('discordHandle' as createdByKeys) },
+	{ label: 'Claimed By', key: ('claimedBy' as bountyKeys) + '.' + ('discordHandle' as claimedByKeys) },
+	{ label: 'Submission Notes', key: ('submissionNotes' as bountyKeys) },
+	{ label: 'Submission URL', key: ('submissionUrl' as bountyKeys) },
+	// { label: 'Paid Status', key: ('paidStatus' as bountyKeys).toString() },
+];
```

Comment:
Trying to add some type checking to the list of keys to be exported. This wasn't actually doing it, other than leaving things blank if the key was wrong, which isn't much of a help. I'd like to see a build error if a key is listed that isn't in the schema.

Reply:
Try this:

```ts
import { BountyCollection } from '@app/models/Bounty';

type NestedDotNotationKeys<T, K extends keyof T & string> = `${K}.${Extract<keyof T[K], string>}`;

type RewardKeys = NestedDotNotationKeys<BountyCollection, 'reward'>;
type CreatedByKeys = NestedDotNotationKeys<BountyCollection, 'createdBy'>;
type ClaimedByKeys = NestedDotNotationKeys<BountyCollection, 'claimedBy'>;

type BountyExportKeys = keyof Omit<BountyCollection, "reward" | "createdBy" | "claimedBy"> | RewardKeys | CreatedByKeys | ClaimedByKeys;

type BountyExportItems = Array<{
	label: string;
	key: BountyExportKeys
}>

export const BOUNTY_EXPORT_ITEMS: BountyExportItems = [
	{ label: 'ID', key: '_id' },
	{ label: 'Title', key: 'title' },
	{ label: 'Description', key: 'description' },
	{ label: 'Criteria', key: 'criteria' },
	{ label: 'Status', key: 'status' },
	{ label: 'Due', key: 'dueAt' },
	{ label: 'Reward Amount', key: 'reward.amount' },
	{ label: 'Reward Currency', key: 'reward.currency' },
	{ label: 'Created', key: 'createdAt' },
	{ label: 'Created By', key: 'createdBy.discordHandle' },
	{ label: 'Claimed By', key: 'claimedBy.discordHandle' },
	{ label: 'Submission Notes', key: 'submissionNotes' },
	{ label: 'Submission URL', key: 'submissionUrl' },
];

export default BOUNTY_EXPORT_ITEMS;
```

Reply:
Could probably try to make a computed type for the whole collection but feels like overkill for one export